pftools/python/parflow/tools/builders.py

```diff
@@ -104,14 +104,16 @@ def side_ids(self, side_patch_ids):
         self.patch_ids_side = side_patch_ids
         return self
 
-    def write(self, name, xllcorner=0, yllcorner=0, cellsize=0, vtk=False):
+    def write(self, name, xllcorner=0, yllcorner=0, cellsize=0, vtk=False, extra=None):
```

Comment:
Is there a reason not to have the default value of extra = [] to simplify the logic slightly?

Reply:
This is a common pattern I've seen in Python for many years (avoid mutable default arguments) to avoid a situation where changes in one invocation of a function/method alter another completely independent invocation. For example:
```
def f(a=[]):
    a.append(42)
    return a

print(f())
print(f())
```
results in:
```
[42]
[42, 42]
```

This happens because default arguments are evaluated **once** when the function is defined, not each time the function is called. Even though this particular method (`write`) is not altering the `extra` argument, its encouraged to avoid having a list as a default argument at all to avoid this gotcha (most IDEs/linters will point this out), should one's implementation get more complex in the future.

Reply:
Thanks for great explanation,  I'm still getting up-to-speed on python.